### PR TITLE
Support non-default `xmlns` and `ConvertFromStringAttribute` in `XamlReader`

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes https://github.com/unoplatform/uno/issues/7536

## PR Type

What kind of change does this PR introduce?

- Bugfix
- Feature

## What is the current behavior?

Not supported.

## What is the new behavior?

- When `XamlReader` loads XAML with non-default `xmlns` with "using:" or "clr-namespace:", it can handle it porperly.
- When string value of a property of type annotated with `ConvertFromStringAttribute` is read, it is properly handled in `XamlReader`.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Markup/Reader/XamlObjectBuilder.cs [hunk 1]
```diff
@@ -822,6 +821,39 @@ private object BuildLiteralValue(XamlMemberDefinition member, Type propertyType
...
 		{
+			if (propertyType.GetCustomAttribute<CreateFromStringAttribute>() is { } createFromString)
+			{
+				var sourceType = propertyType;
+				var methodName = createFromString.MethodName;
+				if (createFromString.MethodName.Contains("."))
+				{
+					var splitIndex = createFromString.MethodName.LastIndexOf(".");
+					var typeName = createFromString.MethodName.Substring(0, splitIndex);
+					sourceType = AppDomain.CurrentDomain
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Markup/Reader/XamlObjectBuilder.cs:832]: TypeResolver should be able to provide this type of lookup

Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Markup/Reader/XamlObjectBuilder.cs:832]: Unfortunately it expects a different format (`ns:Type` or just `Type`), whereas in this case I have full type name

Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Markup/Reader/XamlObjectBuilder.cs]: This approach did not work due to the change I made in this PR - I added this line - https://github.com/unoplatform/uno/blob/47aed3f70f56edcbced643b9119a7489f3cbfd54/src/Uno.UI/UI/Xaml/Markup/Reader/XamlTypeResolver.cs#L359

This covers the scenario where the xmlns of the file is not standard - e.g. xmlns="using:Uno.UI.Tests.Windows_UI_Xaml_Markup.XamlReaderTests which is something that UWP handles properly. This line prepends the default namespace before the type name so "XYZ" becomes "Uno.UI.......XamlReaderTests.XYZ" - which is needed for Type.GetType(name) to work. However, when the type passed into XamlTypeResolver is already fully qualified, it causes it to become "Uno.UI.......XamlReaderTests.Uno.UI.......XamlReaderTests.XYZ" . Not sure how to resolve this to satisfy both scenarios, as it is not really possible to tell whether the string is a qualified or non-qualified type, other than checking if indexof "." is -1.

However, I added this, which makes a fallback to use the original type and that resolves this:

https://github.com/unoplatform/uno/pull/7535/files#diff-9c43ddb90b5f877f348d20114e11c095881eaea4c50d5df26db6f4d359ee2ae5R378

Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Markup/Reader/XamlObjectBuilder.cs]: That works as well!

**Comment by jeromelaban** [comment]
Looks like there are some nullability errors in the XamlReader

**Review by carldebilly — APPROVED** [review]

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**